### ISSUE TITLE
Dedicated control plane subnets

### ISSUE THREAD

**Issue by drmaciej** [terraform-aws-modules/terraform-aws-eks#1312]
## Is your request related to a new offering from AWS?

Not sure if it's very new, but it is supported: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_cluster#subnet_ids
The feature is described as a best practice by AWS at https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html:
> Make sure that the subnets that you specify during cluster creation have enough available IP addresses for the Amazon EKS created network interfaces. We recommend creating small (/28), dedicated subnets for Amazon EKS created network interfaces, and only specifying these subnets as part of cluster creation. Other resources, such as nodes and load balancers, should be launched in separate subnets from the subnets specified during cluster creation.
Currently, the module re-uses subnets across the control plane and workers.

## Describe the solution you'd like.

Ideally, there'd be separate variables for the control plane subnets, and the other subnets.

## Describe alternatives you've considered.

None as of now, the module's structure is preventing me from using this approach.

**Comment by malcolm061990** [comment]
Guys, do you have any plans on this feature?
As I figured out the node groups were created in the same subnets as control plane...
So if control plane is in public subnet then all ec2 instances are in public subnet too but its totally not good :) Yes, it can be configurable via security groups but anyway.
It will be nice to have two separated variables like:
- cluster_subnets
- node_group_subnets

**Comment by daroga0002** [comment]
I taken a look into this and challenge which you are describing here looks be achievable in current version of module. You can specify `subnets` input only for cluster control plane and workers/node groups can be placed in other dedicated subnets via specifying subnets in worker/node groups definition.

as for example:
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/253f927465ccac36bcadefcfb1fd0e39b6244d73/workers.tf#L53

So to summarise I think that probably we don`t require any changes into module itself as this is already allowing such setup.

Does anybody tried to setup this in that way?
Maybe  it will be good to create PR for updating docs and adding example to cover such requirement

**Comment by malcolm061990** [comment]
> I taken a look into this and challenge which you are describing here looks be achievable in current version of module. You can specify `subnets` input only for cluster control plane and workers/node groups can be placed in other dedicated subnets via specifying subnets in worker/node groups definition.
> 
> as for example:
> 
> https://github.com/terraform-aws-modules/terraform-aws-eks/blob/253f927465ccac36bcadefcfb1fd0e39b6244d73/workers.tf#L53
> 
> So to summarise I think that probably we don`t require any changes into module itself as this is already allowing such setup.
> 
> Does anybody tried to setup this in that way? Maybe it will be good to create PR for updating docs and adding example to cover such requirement

You are right. Specifying the subnets in node_group configuration solves the issue. Now the control plane and node_groups can be deployed in separated subnets.
Thanks!

**Comment by antonbabenko** [comment]
This issue has been resolved in version 17.24.0 :tada:
